### DISCUSSION TITLE
EXPIRED LETSENCRYPT CERTIFICATE

### DISCUSSION THREAD

**Discussion by mtran-88** [Ylianst/MeshCentral#6319] [Q&A] [answered]
Hi, I was on vacation and did not get a chance to renew the Letsencrypt certificate, and it expired.  I am getting an error message saying:
![image](https://github.com/user-attachments/assets/266970e9-bae1-4368-b642-97adca0bdad2)

How would I manually renew the Letsencrypt certificate on my meshcentral serer?

Thanks!

**Comment by si458** [comment]
you need to add `"allowedOrigin": "DNS NAME"` under your `domains` sections in config.json
then restart meshcentral and it should autorenew itself

**Reply by mtran-88** [reply]
Hi Simon, thanks for responding!  I still get the same error message when I try to log in.  I've used the exact values you had shown there with the "DNS NAME" and my DNS name typed in with no success.  Note that I am using localhost in the browser on the mc server itself.  
![image](https://github.com/user-attachments/assets/700ce947-ce4c-4683-b492-11c9ab063855)

When I used a remote laptop, and typed in meshcentral.xxxx.com, I get the following error:
![image](https://github.com/user-attachments/assets/cbee1737-3303-4eab-b402-ba950a62e329)

Let me know your thoughts. Thx!

**Reply by si458** [reply]
You should not be accessing the webui using localhost! You should ALWAYS use the dns name! Weather internal or external!
So I'm confused now also, what errors are u exactly getting as the screenshot shows CA error but ur other screen shots the allowedOrigin? So I'm confused?

**Reply by mtran-88** [reply]
Apologies for the confusion!  I thought on the meshcentral server itself, I can use localhost to access meshcentral; it gave me the login screen for meshcentral, and when I enter the username/password, I get that AllowedOrigin error.  If I try to access using the DNS name, I get the CA error.  Hope that is clear.

**Reply by si458** [reply]
Right so yes the allowedOrigin error is because u are using localhost so u should never use localhost but use the dns name instead.

Now why u are still getting the wrong certificate I'm not 100% sure?

One thing u check is stop meshcentral,
Then run it in the full debug mode and check for errors in the output or paste it here and we can have a look see! 
`node node_modules/meshcentral --debug`

Then once it loaded, wait like 5-10secs then access it via private browser and see if it shows any errors/logs that might be helpful!

One other thing to check is the folder structure

You should have folders like meshcentral-data, meshcentral-files, node_modules, and u run the command from here

**Reply by mtran-88** [reply]
Hi, this is what I got when I used the inprivate edge browser and tried to enter the meshcentral.xxxx.com from the meshcentral server. 
C:\mc>node node_modules/meshcentral --debug
MAIN: Core module windows-amt is 621398 bytes.
MAIN: Core module linux-amt is 567321 bytes.
MAIN: Core module linux-noamt is 466785 bytes.
MAIN: Core module windows-recovery is 137824 bytes.
MAIN: Core module linux-recovery is 82212 bytes.
MAIN: Core module windows-agentrecovery is 61977 bytes.
MAIN: Core module linux-agentrecovery is 6365 bytes.
MAIN: Core module windows-tiny is 6305 bytes.
MAIN: Core module linux-tiny is 6305 bytes.
MeshCentral HTTP redirection server running on port 80.
AUTHLOG: Server listening on 0.0.0.0 port 80.
CERT: LE: Getting certs from local store (Production)
CERT: LE: Reading certificate files
CERT: LE: Setting LE cert for default domain.
MeshCentral v1.1.22, Hybrid (LAN + WAN) mode.
DISPATCH: AddEventDispatch [ 'server-shareremove' ]
DISPATCH: AddEventDispatch [ '*' ]
DISPATCH: DispatchEvent [ '*' ]
MAIN: Server started
MeshCentral Intel(R) AMT server running on meshcentral.xxxx.com:4433.
AUTHLOG: Server listening on 0.0.0.0 port 4433.
HTTPS: Server listening on 0.0.0.0 port 443.
MeshCentral HTTPS server running on meshcentral.xxxx.com:443.
MPS: New CIRA connection
CERT: LE: Certificate has 89 day(s) left.
CERT: LE: Certificate is ok.
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: New CIRA connection
MPS: CIRA timeout, disconnecting.
MPS: New CIRA connection
MPS: New CIRA connection
MPS: CIRA connection error Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'read'
}
MPS: New CIRA connection

I got this afterwards:
![image](https://github.com/user-attachments/assets/8a6e8c5b-903d-42e1-9669-93bbdd90d047)

**Reply by si458** [reply, marked as answer]
That's the wrong web ui! Thats the intel AMT port!

Something is misconfigured with ur firewall/meshcentral

U should leave port as 443 in config.json
And set aliasport to also 443 (unless u want to access it on a different port like `https://meshcentral.mydomain.com:1234` then set it as 1234 for example)

Then u need to set port 80 to 80
And set 443 to 443 in ur firewall
Or 1234 to 443 if u wanted a different external port

DONT USE/SET PORT 4433 ANYWHERE!

Remember to stop the meshcentral after u change the config.json and restart it

**Reply by mtran-88** [reply]
Thanks Simon!  It was the firewall as you suspected!

**Reply by si458** [reply]
Glad it's sorted!
P.S don't forget to donate to help the project alive 🧡 https://www.si458.co.uk/2024/01/05/donation/
